main.go

```diff
@@ -217,6 +217,7 @@ func main() {
 		Scheme: scheme,
 		Metrics: metricsserver.Options{
 			BindAddress:    metricsBindAddr,
+			SecureServing:  true,
 			FilterProvider: filters.WithAuthenticationAndAuthorization,
```

Comment:
```suggestion
			FilterProvider: filters.WithAuthenticationAndAuthorization,
			TLSOpts:       tlsOpts, 
```

Reply:
Also, please ensure that you disable HTTP/2 by default : https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4/cmd/main.go#L95-L109

The fix for GRCP are not good enough: https://github.com/kubernetes/kubernetes/issues/121197